pkg/client/unversioned/restclient.go

```diff
@@ -45,26 +44,24 @@ const (
 //
 // Most consumers should use client.New() to get a Kubernetes API client.
 type RESTClient struct {
-	baseURL *url.URL
-	// A string identifying the version of the API this client is expected to use.
-	groupVersion unversioned.GroupVersion
+	// Base is the root URL for all invocations of the client
+	Base *url.URL
```

Comment:
We're already violating the spirit of closed class here - the 2e tests use both Client and throttle and perform operations that really don't belong there (the e2e tests also assume client is set, which is incorrect).  I'll revert the public part of this and remove the usage of Codec from rolling update (which didn't need to be doing this).